### ISSUE TITLE
Missing Windows compatibility

### ISSUE THREAD

**Issue by Flimm** [Flimm/backup-all-my-flickr-photos#2]
There a couple of known issues for Windows compatibility:
- [x] This script will attempt to create files with invalid filenames on Windows, for instance `\`.
- [x] This script runs `file`, which is not available on Windows AFAIK.

**Comment by hombit** [comment]
The first problem for `\` is solved by ec4ca577fd61d38f5b43d445c3a0a9a379d1e0ff.

**Comment by hombit** [comment]
Use http headers mime-type instead of running `file` (commit db96ed892c98fc9e477ff979b3fd51e7a88c2692)

**Comment by hombit** [comment]
I decided to delete forbidden in Windows characters for every system. The reason is that a user can use `NTFS` with Unix or use `Samba` to share backup with Windows or copy photos to Windows. So the first problem is completely solved by 1d63b83cc2b27bc780474114b6fb59190bd69144

**Comment by Flimm** [comment]
Sweet! The problem looks fixed. I don't think the second problem probably is fixed, I recall that Flickr frequently set the wrong `Content-Type` header for videos, which is why I had to resort to running `file`.

**Comment by hombit** [comment]
@Flimm Can we extract extension from url returned by `photo.getPhotoFile()`? I haven't videos in my account so I cannot check what url they have, but for images it ends with '.jpg'

**Comment by hombit** [comment]
@Flimm Can you give an example of the wrong `Content-Type` header?

**Comment by Flimm** [comment]
Indeed. I uploaded an Apple Quicktime file with a `.mov` extension, but Flickr serves it with a `video/mp4` mime-type. Here's [the sample `.mov` file](http://samples.mplayerhq.hu/mov/rawbgr24.mov) I used.

**Comment by hombit** [comment]
Ok. I propose the fallowing steps to determine extension:
1) Look for url returned by `photo.getPhotoFile()`. It can contain or not extension. For example, url of one of my images is https://farm9.staticflickr.com/8605/28386221452_0efc1cbefa_o_d.jpg but for this video I've got https://www.flickr.com/photos/hombit/28888981875/play/orig/7da54fff22/. If url's path has an extension than we use it.
2) Sometimes there is `Content-Disposition` header with `filename` field. We can use it if `filename` has an extension. For example for this video I have `Content-Disposition: attachment; filename=28888981875.mov`.
3) Use `Content-Type` header.
4) If nothing helped then use `.jpg` extension

I'll code this if you agree with this behaviour.

**Comment by Flimm** [comment]
I implemented your suggestion, except for step 3 and step 4. I would rather throw an exception than give it a wrong extension. Here's the commit: https://github.com/Flimm/backup-all-my-flickr-photos/commit/e224bb3a6f8af46ae65158e248e826d756961cfe

I also tried using `python-magic` in [another branch](https://github.com/Flimm/backup-all-my-flickr-photos/tree/magic).

I've tested this a bit, I would like to test it on a bigger Flickr account before releasing it.

Thanks for the tip about `Content-Disposition`, it seems that header is set for video files!
